### PR TITLE
doc improvements: api examples, event notifications

### PR DESCRIPTION
**Pull request by opqdonut**
related to #2571

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue

## Documentation
- [x] Update docs/ (if applicable)

### CODE REVIEW

**Review by marharyta — APPROVED** [review]

**Comment by opqdonut** [comment]
waiting for feedback from @jaakkocsc before merging

**Pull request merged**
